src/store/actions/getNFTAssets.ts

```diff
@@ -0,0 +1,82 @@
+import { ActionContext, rootActionContext } from '..';
+import { Asset, Network, NFTAsset, WalletId } from '../types';
+
+export const getNFTAssets = async (
+  context: ActionContext,
+  {
+    network,
+    walletId,
+    asset,
+    accountId
+  }: {
+    network: Network;
+    walletId: WalletId;
+    asset: Asset;
+    accountId: string;
+  }
+): Promise<any> => {
+  const { state, commit, getters } = rootActionContext(context);
+  const client = getters.client({
+    network,
+    walletId,
+    asset,
+  });
+
+  const nft = await client.nft.fetch();
+
+  const account =  state.accounts[walletId]![network].find((a) => a.id === accountId);
+
+  const nftAssetsStoredInState: NFTAsset[] = account?.nftAssets || [];
+  const nftAssetsFetched: NFTAsset[] = nft.assets;
+
+  // set new nft assets to a new map
```

Comment:
I see that you added comments but we really need to simplify this function overall
It looks like it's jsut merging 2 arrays and default starred = false. That shouldn't require this much code and should be doable with some map+reduce. Can you help @kraikov ?

Reply:
I think we can simply use the fetched array and set the starred ones from the state.

Reply:
Thanks, @monokh @kraikov. 👍 
This has been addressed.